transactor/src/main/scala/io/mediachain/datastore/DynamoDatastore.scala

```diff
@@ -0,0 +1,158 @@
+package io.mediachain.datastore
+
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.amazonaws.services.dynamodbv2.model.AttributeValue
+import java.nio.ByteBuffer
+import scala.collection.mutable.{Buffer, ArrayBuffer}
+import io.mediachain.multihash.MultiHash
+import io.mediachain.types.Datastore.DatastoreException
+
+// TODO error handling
+//      queued eventual writes in the background with disk backing of 
+//       in-progress writes
+//      use batch writes for chunked puts
+class DynamoDatastore(table: String, creds: BasicAWSCredentials)
+  extends BinaryDatastore with AutoCloseable {
+  import scala.collection.JavaConversions._
+  
+  val chunkSize = 1024 * 384 // 384 KB; DynamoDB has 400KB limit
+  val chunkTable = table + "Chunks"
+  
+  val db = new AmazonDynamoDBClient(creds)
+
+  override def put(key: MultiHash, value: Array[Byte]) {
+    if (value.length < chunkSize) {
+      putSimple(key, value)
+    } else {
+      putChunked(key, value)
+    }
+  }
+
+  private def putSimple(key: MultiHash, value: Array[Byte]) {
+    val keyAttr = new AttributeValue
+    keyAttr.setS(key.base58)
+    
+    val dataAttr = new AttributeValue
+    dataAttr.setB(bytes2Buffer(value))
+    
+    val item = Map(
+      "multihash" -> keyAttr,
+      "data"      -> dataAttr
+    )
+    
+    db.putItem(table, item)
+  }
+
+  private def putChunked(key: MultiHash, value: Array[Byte]) {
+    val key58 = key.base58
+    val chunks = chunkCount(value.length)
+    val chunkIds = (1 to chunks).map { n => key58 + "#" + n }
+    
+    // this should use the BatchWriteItemRequest API (PITA alert)
+    chunkIds.zipWithIndex.foreach {
```

Comment:
no, the intention is to have the PersistentStore that does queued writes retry the entire put if any chunk fails.